### ISSUE TITLE
Duplicate assignment-check fails

### ISSUE THREAD

**Issue by emipa606** [MSEndpointMgr/IntuneWin32App#101]
There is an issue with the `Test-IntuneWin32AppAssignment.ps1` 
It seems the method just checks if there are any assignments on the specified group but does not take into account what type of assignment it is. The following setup is a valid use-case that is blocked by this test:

- Required Installation on the group `Install - Application`
- Required Uninstallation on the group `Uninstall - Application`
- Available Install on the generic group `All available applications`
- Exclusion from Available Install on the group `Uninstall - Application`

In the above the last step is blocked as there already is an assignment on the `Uninstall - Application` group. However, the exclusion of this group is to not show the application as available for the members of the forced uninstall-group. This is possible to do in the web-gui as well. As the Company Portal does not uninstall applications that is listed as available for a user, this exclusion is the only way to be sure that the application is correctly uninstalled when added to `Uninstall - Application`

There is another use-case where having multiple assignments on the same group is valid. This would be where you add the `Uninstall - Application` as a Exclusion on the required Install-assignment to be sure that noone can get both the install and uninstall-assignment a the same time.

I suggest that the check needs to take Exclusion/Inclusion into account when checking for duplicate assignments.

Reference:
https://github.com/MSEndpointMgr/IntuneWin32App/blob/c898faacaf7f1d660ec3ecf178b779c7f8505669/Private/Test-IntuneWin32AppAssignment.ps1#L59-L66

**Comment by emipa606** [comment]
Duplicate of #76 
Adding info in that instead
